src/Compilers/CSharp/Portable/Symbols/Source/SourceNamedTypeSymbol_Bases.cs

```diff
@@ -640,5 +712,93 @@ private NamedTypeSymbol MakeAcyclicBaseType(DiagnosticBag diagnostics)
 
             return declaredBase;
         }
+
+        private TypeSymbol MakeAcyclicExtensionClassType(DiagnosticBag diagnostics)
+        {
+            if (!this.IsExtensionClass)
```

Comment:
`NamedTypeSymbol.ExtensionClassType` is documented to return null if it is not an extension class, and this is where that behavior is implemented for source type symbols. Asserting would be inconsistent with other implementations of ExtensionClassType (e.g. what happens if you call ExtensionClassType on a TupleTypeSymbol or SynthesizedContainer), as well as BaseType (which has the same documented behavior of returning null on a type that doesn't have a base type).